app/helpers/annotations_helper.rb

```
@@ -0,0 +1,99 @@
+    
+    # Add file-specific annotations for this problem
+    annotations_by_file_data.each do |_, file_annotations|
+      file_annotations.each do |annotation_data|
+        id = annotation_data[4]
+        annotation = Annotation.find_by(id: id)
+        problem_annotations << annotation if annotation && annotation.problem_id == problem_id
+      end
+    end
+    
+    problem_annotations.compact
+  end
```

RuboCop (1.73)

[convention] Cyclomatic complexity for `get_problem_annotations` is too high. [9/7]

(Metrics/CyclomaticComplexity)